pilot/pkg/networking/v1alpha3/listener.go

```diff
@@ -45,3 +83,532 @@ func BuildListeners(env model.Environment, node model.Proxy) ([]*xdsapi.Listener
 	}
 	return nil, nil
 }
+
+// buildSidecarListeners produces a list of listeners for sidecar proxies
+func buildSidecarListeners(
+	mesh *meshconfig.MeshConfig,
+	proxyInstances []*model.ServiceInstance,
+	services []*model.Service,
+	managementPorts model.PortList,
+	node model.Proxy,
+	config model.IstioConfigStore) []*xdsapi.Listener {
+
+	// ensure services are ordered to simplify generation logic
+	sort.Slice(services, func(i, j int) bool { return services[i].Hostname < services[j].Hostname })
+
+	listeners := make([]*xdsapi.Listener, 0)
+
+	//if node.Type == model.Router {
+	//	outbound := buildOutboundListeners(mesh, node, proxyInstances, services, config)
+	//	listeners = append(listeners, outbound...)
+	//} else
+	if mesh.ProxyListenPort > 0 {
+		inbound := buildInboundListeners(mesh, node, proxyInstances, config)
+		outbound := buildOutboundListeners(mesh, node, proxyInstances, services, config)
+		// TODO: health check ports
+		//mgmtListeners := buildMgmtPortListeners(mesh, managementPorts, node.IPAddress)
+
+		listeners = append(listeners, inbound...)
+		listeners = append(listeners, outbound...)
+
+		// If management listener port and service port are same, bad things happen
+		// when running in kubernetes, as the probes stop responding. So, append
+		// non overlapping listeners only.
+		//for i := range mgmtListeners {
+		//	m := mgmtListeners[i]
+		//	l := getByAddress(listeners, m.Address.String())
+		//	if l != nil {
+		//		log.Warnf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
+		//			m.Name, m.Address, l.Name, l.Address)
+		//		continue
+		//	}
+		//	listeners = append(listeners, m)
+		//}
+
+		// We need a dummy filter to fill in the filter stack for orig_dst listener
```

Comment:
We can remove this once the `original_dst` filter is ready, right? Maybe make a note of it here in the comment as a TODO.

Reply:
See "use_original_dst" field here: https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/lds.proto.html#envoy-api-msg-listener